melodies_monet/plots/aircraftplots.py

```diff
@@ -0,0 +1,842 @@
+    elif domain_type == 'epa_region' and domain_name is not None:
+        latmin,lonmin,latmax,lonmax,acro = get_epa_bounds(index=None,acronym=domain_name)
+        plt.title('EPA Region ' + domain_name + ': ' + label_m + ' - ' + label_o,fontweight='bold',**text_kwargs)
+    else:
+        latmin= math.floor(min(df.latitude))
+        lonmin= math.floor(min(df.longitude))
+        latmax= math.ceil(max(df.latitude))
+        lonmax= math.ceil(max(df.longitude))
+        plt.title(domain_name + ': ' + label_m + ' - ' + label_o,fontweight='bold',**text_kwargs)
+    
+    if 'extent' not in map_kwargs:
+        map_kwargs['extent'] = [lonmin,lonmax,latmin,latmax]  
+    ax.axes.set_extent(map_kwargs['extent'],crs=ccrs.PlateCarree())
+    
+    #Update colorbar
+    f = plt.gcf()
+    model_ax = f.get_axes()[0]
+    cax = f.get_axes()[1]
+    #get the position of the plot axis and use this to rescale nicely the color bar to the height of the plot.
+    position_m = model_ax.get_position()
+    position_c = cax.get_position()
+    cax.set_position([position_c.x0, position_m.y0, position_c.x1 - position_c.x0, (position_m.y1-position_m.y0)*1.1])
+    cax.set_ylabel(ylabel,fontweight='bold',**text_kwargs)
+    cax.tick_params(labelsize=text_kwargs['fontsize']*0.8,length=10.0,width=2.0,grid_linewidth=2.0)    
+    
+    #plt.tight_layout(pad=0)
+    savefig(outname + '.png', loc=4, logo_height=120)
+    
+####NEW function for adding 'altitude' variable as secondary y- axis (qzr++)
+def add_yax2_altitude(ax, pairdf, altitude_variable, altitude_ticks, text_kwargs): 
+    """Creates secondary y-axis (altitude) for timeseries plot.
+    
+    Parameters
+    ----------
+    ax : ax
+        Matplotlib ax from previous occurrences so it can overlay obs and model results on the same plot.
+    pairdf : pandas.DataFrame
+        Model/obs paired data to plot.
+    text_kwargs : dictionary
+        Dictionary containing information about text.
+    altitude_variable : str
+        Altitude variable in th paired df to be plotted on the secondary y-axis,  e.g., 'MSL_GPS_Altitude_YANG'
+    altitude_ticks : float or int
+        Altitude tick interval in meters for the secondary y-axis, for altitude (m) for instance, can be for pressure (if chosen as altitude_variable) 
+                
+    Returns
+    -------
+    ax : ax
+        Matplotlib ax such that driver.py can iterate to overlay multiple models on the same plot.
+    """
+    ax2 = ax.twinx()
+    ax2.plot(pairdf.index, pairdf[altitude_variable], color='g', label='Altitude (m)')
+    ax2.set_ylabel('Altitude (m)', color='g', fontweight='bold', fontsize=text_kwargs['fontsize'])
+    ax2.tick_params(axis='y', labelcolor='g', labelsize=text_kwargs['fontsize'] * 0.8)
+    ax2.set_ylim(0, max(pairdf[altitude_variable]) + altitude_ticks)
+    ax2.set_xlim(ax.get_xlim())
+    ax2.yaxis.set_ticks(np.arange(0, max(pairdf[altitude_variable]) + altitude_ticks, altitude_ticks))
```

Comment:
Like my prior comment, I think this should be controlled by `vmin2` and `vmax2`. I think it's potentially fine to use the current functionality as the default but if a user wants to change the secondary yaxis limits directly they should be able to.